internal/engine/execution.go

```diff
@@ -150,9 +150,10 @@ func (e *Engine) ExecuteAndRenderSteps(steps []Step, env map[string]string) erro
 							actualOutput := commandOutput.StdOut
 							expectedOutput := block.ExpectedOutput.Content
 							expectedSimilarity := block.ExpectedOutput.ExpectedSimilarity
+							expectedContains := block.ExpectedOutput.ExpectedContains
 							expectedOutputLanguage := block.ExpectedOutput.Language
 
-							outputComparisonError := compareCommandOutputs(actualOutput, expectedOutput, expectedSimilarity, expectedOutputLanguage)
+							outputComparisonError := compareCommandOutputs(actualOutput, expectedOutput, expectedSimilarity, expectedContains, expectedOutputLanguage)
```

Comment:
If we continue adding new matching mechanism, we should consider passing directly the whole `ExpectedOutput` struct.